### ISSUE TITLE
Update ReSpec to support oslc-op Project Specification rendering

### ISSUE THREAD

**Issue by jamsden** [oslc-op/OSLC-admin#6]
Specific changes needed are based on the proposed Project Specification template:

- [ ] Styles
Don't know what the changes are at this point

- [ ] Banner
￼Update the banner to the new OASIS Open Project banner.
￼
- [ ] Document specStatus
Committee Specification --> Project Specification
```
WD
CSD -> PSD
CSPRD -> PSPRD
CS -> PS
COS
OS
Errata
CND -> PND
CNPRD -> PNPRD
CN -> PN
```

- [ ] This, Previous, Latest versions:
Are there any changes in the naming convention or URLs? GitHub contains approved, normative work products while docs.oasis-open.org is a copy/backup that will need to align with GitHub URLs. We can sort this out once the repo is up, the resources are migrated, and we determine what it means to publish.

For OSLC, the GitHub repo will contain the ReSpec source. Clicking on a GitHub URL will automatically run ReSpec to produce the published results. So as long as a combination of the source and ReSpec generation is correct for the HTML, then there will be nothing to do to publish OSLC OP specs. 

However, storage of the static HTML, generation of PDF files, and adding page footers would be another matter. We could treat that publish process the same as we do now with the results published on docs.oasis-open.org using whatever naming convention OASIS wishes to do.

The question is, who does this publishing? Hopefully it is OASIS, not the OP TSC.

- [ ] Technical Committee wG, wgShortName, wgURI, wgPublicList
Technical Committee: --> Open Project:

- [ ] Chairs
Chairs: --> Project Chairs:
And does this need to be a PGB member or can TSC be a chair?

PGB and TSC both need a chair or co-chairs. PGB should have a set for a member of the TSC, but it need not be the chair. 

- [ ] Additional content:
Additional artifacts: --> Additional components:

- [ ] Status
completely changed:

This specification is published under the [Apache License Version 2.0; Eclipse Public License – v 1.0; The 3-Clause BSD License; Attribution 2.0 (CC BY 2.0); Attribution 4.0 International (CC BY 4.0); The MIT License]. 

[Optional if needed:  Portions of this specification also provided under the <additional applicable license names and URLs>.] [Do we need to include this:  For information on whether any patents have been disclosed that may be essential to implementing this specification, and any offers of patent licensing terms, please refer to the Intellectual Property Rights notification section of the Project (https://www.oasis-open.org/committees/OP-short-name/ipr.php). ]

This specification is published under the Apache License Version 2.0. 

Is this needed? For information on whether any patents have been disclosed that may be essential to implementing this specification, and any offers of patent licensing terms, please refer to the Intellectual Property Rights notification section of the Project (https://www.oasis-open.org/committees/OP-short-name/ipr.php).

- [ ] Citation format
OASIS Committee Specification --> OASIS Project Specification

- [ ] Notices
Additional paragraph added after Copyright: for licensing:

This work is licensed under the the [Apache License Version 2.0; Eclipse Public License – v 1.0; The 3-Clause BSD License; Attribution 2.0 (CC BY 2.0); Attribution 4.0 International (CC BY 4.0); The MIT License]. [Optional: Portions of this specification also provided under the <additional applicable license names and URLs>.] All contributions made to this project have been made under the OASIS Contributor License Agreement (CLA).

This work is licensed under the the Apache License Version 2.0. All contributions made to this project have been made under the OASIS Contributor License Agreement (CLA).

- [ ] Introduction
IPR policy removed, replaced with license paragraph in Notices section
Separate sections for normative and non-normative references not sub-sections of References?

**Comment by ndjc** [comment]
ReSpec now has the changes for the checked items in the issue description above.
Still pending: the exact URI rules. Following on from the discussion in last week's meeting and Andrew's suggestions recorded in the [minutes of that meeting](https://github.com/oslc-op/oslc-admin/blob/master/minutes/2019/2019-06-20.md), I have created a wiki page https://github.com/oslc-op/oslc-specs/wiki/URIs-in-Specifications. Please review that page as soon as possible so I can complete the updates to ReSpec.

**Comment by jamsden** [comment]
I see a few things we might want to discuss:

- [ ] open-services.net/spec doesn't match what's currently used: open-services.net/specifications

- [ ] spec-short-name (e.g., osld-rm) doesn't match the repo folder: specs/rm, it might be helpful if it did, for relative cross document references if any, and for consistency

- [ ] The URIs are missing the document status: e.g., `https://open-services.net/spec/spec-short-name/revision-label/spec-status/document-name.suffix`

- [ ] document-name will break relative links across multi-part specifications. I have been using the file names. 

- [ ] having all the open-services.net (or other) spec URIs redirect to GitHub will cause them to be re-rendered every time and doesn't work for PDFs (which aren't subject to content negotiation). 

- [ ] I think it might be useful to include the edDraftURL in the documents themselves. This could be automatically created by ReSpec, just like thisVersion and latestVersion. 

- [ ] "Publishing" a specification would consist of:

1. Labeling the version in the GitHub repo
2. Creating a new `https://open-services.net/spec/spec-short-name/revision-label/spec-status/document-name.suffix' redirect at open-services.net
3. Adjusting the `https://open-services.net/spec/spec-short-name/document-name.suffix` to point to the newly published revision
4. Exporting the HTML using ReSpec
5. Generating the PDF
6. Creating a new `https://open-services.net/spec/spec-short-name/revision-label/spec-status/document-name.pdf' document open-services.net
7. Adjusting the `https://open-services.net/spec/spec-short-name/document-name.pdf` to point to the newly published PDF revision

Anything else?

**Comment by ndjc** [comment]
Let's discuss in the meeting. I had started to write a detailed response, but then accidentally closed the window (to which Github did not object - poor usability!), and it is now far too late at night (well, early in the morning, actually) for me to retype it all.

**Comment by jadelkhoury** [comment]
I will be joining 15 min late into the meeting, due to a meeting clash

**Comment by berezovskyi** [comment]
@chet-ensign could we move the Notices section to the end of the specs?

**Comment by berezovskyi** [comment]
I would also like to see references at the end of the spec.

**Comment by berezovskyi** [comment]
@jamsden the publication process should be adjusted to set the `publishDate` and  `specStatus` and `revision` in a PR, reviewed by the maintainers. Once a PR is reviewed, PGB ballot is opened for 14 days to approve a PSD. Once the PSD approves, the PR is merged the commit is tagged `% specStatus%-%revision%` and another PR is merged to IMMEDIATELY reverse the PS/PSD attributes, returning the spec on the master branch to the WD status and dropping or bumping the revision. 

Not having such a process causes us https://github.com/oslc-op/oslc-specs/issues/61 and https://github.com/oslc-op/oslc-specs/issues/62.

This is almost the exact process we have for Lyo releases, dropping the SNAPSHOT and then immediately bumping the version and adding the SNAPSHOT back to prevent overwriting the released artefacts on the Maven repo.
